### PR TITLE
Docs: Document support for binary in truncate transform

### LINKED ISSUE #5251: Discrepancy between partition truncation transform in spec vs code (org.apache.iceberg.transforms.Transform)
I would like some clarification on wether or not Iceberg officially supports partition truncation transform for Fixed or Binary data types.

The [specification](https://iceberg.apache.org/spec/#partition-transforms) denotes that the only data types supported for truncate transform are: String, Int, Long, and Decimal. Yet the following [code](https://github.com/apache/iceberg/blob/5009949ba4377ac5a8572ff7ae70e886c9e33bec/api/src/main/java/org/apache/iceberg/transforms/Truncate.java#L37) seems to suggest that fixed/binary data types might be supported as well:

 static <T> Truncate<T> get(Type type, int width) {
    Preconditions.checkArgument(width > 0,
        "Invalid truncate width: %s (must be > 0)", width);

    switch (type.typeId()) {
      case INTEGER:
        return (Truncate<T>) new TruncateInteger(width);
      case LONG:
        return (Truncate<T>) new TruncateLong(width);
      case DECIMAL:
        return (Truncate<T>) new TruncateDecimal(width);
      case STRING:
        return (Truncate<T>) new TruncateString(width);
      case BINARY:
        return (Truncate<T>) new TruncateByteBuffer(width);
      default:
        throw new UnsupportedOperationException(
            "Cannot truncate type: " + type);
    }
  }

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Fixes #5251

It seems unlikely we will address this discrepancy by dropping support for binary. I'd like to propose documenting the currently implemented behavior instead.

### CODE REVIEW

File: format/spec.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ Partition field IDs must be reused if an existing partition spec contains an equ
 |-------------------|--------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------|-------------|
 | **`identity`**    | Source value, unmodified                                     | Any                                                                                                       | Source type |
 | **`bucket[N]`**   | Hash of value, mod `N` (see below)                           | `int`, `long`, `decimal`, `date`, `time`, `timestamp`, `timestamptz`, `timestamp_ns`, `timestamptz_ns`, `string`, `uuid`, `fixed`, `binary` | `int`       |
-| **`truncate[W]`** | Value truncated to width `W` (see below)                     | `int`, `long`, `decimal`, `string`                                                                        | Source type |
+| **`truncate[W]`** | Value truncated to width `W` (see below)                     | `int`, `long`, `decimal`, `string`, `binary`                                                              | Source type |
```
Comment (amogh-jahagirdar) [inline format/spec.md:317]: I think we need to explicitly define that truncate(binary, width) essentially is a subarray from (0, width) similar to what  we do for truncate(string)

File: format/spec.md [hunk 2]
```diff
@@ -245,19 +245,19 @@ For example, a file may be written with schema `1: a int, 2: b string, 3: c doub
...
 
-* `names`: A required list of 0 or more names for a field. 
+* `names`: A required list of 0 or more names for a field.
 * `field-id`: An optional Iceberg field ID used when a field's name is present in `names`
 * `fields`: An optional list of field mappings for child field of structs, maps, and lists.
 
 Field mapping fields are constrained by the following rules:
 
-* A name may contain `.` but this refers to a literal name, not a nested field. For example, `a.b` refers to a field named `a.b`, not child field `b` of field `a`. 
-* Each child field should be defined with their own field mapping under `fields`.
```
Comment (amogh-jahagirdar) [inline format/spec.md:255]: If possible, can we undo all the changes which are space removals at the end of the line? It'll be easier to review the PR that way

File: format/spec.md (hunk 1 above)
Comment (amogh-jahagirdar) [reply format/spec.md:317]: Sorry I meant specifically adding an entry to this table https://github.com/apache/iceberg/pull/10079/files#diff-36347a47c3bf67ea2ef6309ea96201814032d21bb5f162dfae4045508c15588aR353

Comment (TheNeuralBit) [reply format/spec.md:317]: Done! Please check my language and example

File: format/spec.md (hunk 2 above)
Comment (TheNeuralBit) [reply format/spec.md]: Done

**Review by TheNeuralBit — COMMENTED** [review]
Thanks for the feedback :)

**Review by Fokko — APPROVED** [review]
Looks good. We already do this for PyIceberg as well: https://github.com/apache/iceberg-python/blob/ee4dd92008ffa21067837fe9fcdfaea063a514f3/pyiceberg/transforms.py#L633-L635

**Review by rdblue — APPROVED** [review]

**Review by amogh-jahagirdar — APPROVED** [review]

**Comment by amogh-jahagirdar** [comment]
Merged. Thanks for addressing this @TheNeuralBit! thanks for the reviews @Fokko @rdblue

**Pull request merged**